### PR TITLE
Group Structures for Complex Surface Representations

### PR DESCRIPTION
**Pull request by mjohnson541**
This PR primarily focuses on the addition of the Rx and Rx!H atomtypes along with a few bugfixes.

### CODE REVIEW

File: rmgpy/molecule/group.py [hunk 1]
```diff
@@ -806,9 +806,9 @@ def get_order_str(self):
                 values.append('B')
             elif value == 0:
                 values.append('vdW')
-            elif value == 0.1:
+            elif abs(value - 0.1) < 1e-4:
```
Comment (hwpang) [inline rmgpy/molecule/group.py:809]: Why is this needed?

File: rmgpy/molecule/group.py [hunk 2]
```diff
@@ -940,6 +940,23 @@ def is_van_der_waals(self, wildcards=False):
         else:
             return abs(self.order[0]) <= 1e-9 and len(self.order) == 1
 
+            
+    def is_reaction_bond(self, wildcards=False):
```
Comment (hwpang) [inline rmgpy/molecule/group.py:944]: Can you add a test for this function

File: rmgpy/molecule/group.py [hunk 3]
```diff
@@ -1375,10 +1392,13 @@ def clear_reg_dims(self):
             atm.reg_dim_atm = [[], []]
             atm.reg_dim_u = [[], []]
             atm.reg_dim_r = [[], []]
+            atm.reg_dim_site = [[],[]]
+            atm.reg_dim_morphology = [[],[]]
         for bd in self.get_all_edges():
             bd.reg_dim = [[], []]
 
-    def get_extensions(self, r=None, basename='', atm_ind=None, atm_ind2=None, n_splits=None):
+    def get_extensions(self, r=None, r_bonds=[1,2,3,1.5,4], r_un=[0,1,2,3], basename='', atm_ind=None, atm_ind2=None, n_splits=None):
```
Comment (hwpang) [inline rmgpy/molecule/group.py:1400]: Can you set the defaults to `None` and set them within the function? It's bad practice to have mutable defaults.

Comment (hwpang) [reply rmgpy/molecule/group.py:1400]: Can you add a test for this function too?

**Review by hwpang — CHANGES_REQUESTED** [review]
Thanks for the PR! Could you add tests for these changes? We should really catch these type of bugs in tests.

File: rmgpy/molecule/group.py (hunk 1 above)
Comment (mjohnson541) [reply rmgpy/molecule/group.py:809]: Sometimes floating point errors from incrementing bond orders result in values for the fraction bond orders that are not quite 0.1 or not quite 0.05.

File: rmgpy/molecule/group.py (hunk 3 above)
Comment (mjohnson541) [reply rmgpy/molecule/group.py:1400]: Changed. This function has a test already.

File: rmgpy/molecule/group.py (hunk 2 above)
Comment (mjohnson541) [reply rmgpy/molecule/group.py:944]: added

File: rmgpy/molecule/atomtype.py [hunk 4]
```diff
@@ -240,20 +240,56 @@ def get_features(self):
 
 ATOMTYPES = {}
 
+ATOMTYPES['Rx']    = AtomType(label='Rx', generic=[], specific=[
```
Comment (hwpang) [inline rmgpy/molecule/atomtype.py:243]: Can you run the changes through a formatter

File: rmgpy/molecule/group.py [hunk 5]
```diff
@@ -1387,7 +1406,10 @@ def get_extensions(self, r=None, basename='', atm_ind=None, atm_ind2=None, n_spl
                        extents=list, RnH=list, typ=list)
 
         extents = []
-
+        if r_bonds is None:
+            r_bonds=[1,2,3,1.5,4]
```
Comment (hwpang) [inline rmgpy/molecule/group.py:1410]: Can you run a formatter through the changes

File: test/rmgpy/molecule/groupTest.py [hunk 6]
```diff
@@ -1625,3 +1625,21 @@ def test_get_element_count(self):
...
         assert expected == result
+    
+    def test_is_reaction_bond(self):
+        """
+        Test the is_reaction_bond() function
+        """
+        group = Group().from_adjacency_list("""
+                                            1 R!H u1 {2,R} {3,vdW}
+                                            2 R u0 {1,R}
+                                            3 H u0 {vdW,1} {4,S}
```
Comment (hwpang) [inline test/rmgpy/molecule/groupTest.py:1636]: Why `{vdW,1}` but not `{1,vdW}`?

```suggestion
                                            3 H u0 {vdW,1} {4,S}
```

**Review by hwpang — CHANGES_REQUESTED** [review]
Some questions and request for formatting

File: rmgpy/molecule/atomtype.py (hunk 4 above)
Comment (mjohnson541) [reply rmgpy/molecule/atomtype.py]: I've made the changes I think you're looking at, at least.

File: rmgpy/molecule/group.py (hunk 5 above)
Comment (mjohnson541) [reply rmgpy/molecule/group.py]: I've made the changes I think you're looking at, at least.

File: test/rmgpy/molecule/groupTest.py (hunk 6 above)
Comment (mjohnson541) [reply test/rmgpy/molecule/groupTest.py]: Fixed

File: rmgpy/molecule/group.py [hunk 7]
```diff
@@ -940,6 +940,23 @@ def is_van_der_waals(self, wildcards=False):
...
+    def is_reaction_bond(self, wildcards=False):
+        """
+        Return ``True`` if the bond represents a van der Waals bond or ``False`` if
+        not. If `wildcards` is ``False`` we return False anytime there is more
+        than one bond order, otherwise we return ``True`` if any of the options
+        are van der Waals.
+        """
+        if wildcards:
+            for order in self.order:
+                if abs(order[0]-0.05) <= 1e-9:
```
Comment (hwpang) [inline rmgpy/molecule/group.py:953]:
```suggestion
                if abs(order[0] - 0.05) <= 1e-9:
```

File: rmgpy/molecule/group.py [hunk 8]
```diff
@@ -940,6 +940,23 @@ def is_van_der_waals(self, wildcards=False):
...
+                    return True
+            else:
+                return False
+        else:
+            return abs(self.order[0]-0.05) <= 1e-9 and len(self.order) == 1
```
Comment (hwpang) [inline rmgpy/molecule/group.py:958]:
```suggestion
            return abs(self.order[0] - 0.05) <= 1e-9 and len(self.order) == 1
```

File: rmgpy/molecule/group.py [hunk 9]
```diff
@@ -1387,7 +1406,10 @@ def get_extensions(self, r=None, basename='', atm_ind=None, atm_ind2=None, n_spl
                        extents=list, RnH=list, typ=list)
 
         extents = []
-
+        if r_bonds is None:
+            r_bonds = [1,2,3,1.5,4]
```
Comment (hwpang) [inline rmgpy/molecule/group.py:1410]:
```suggestion
            r_bonds = [1, 1.5, 2, 3, 4]
```

File: rmgpy/molecule/group.py [hunk 10]
```diff
@@ -1387,7 +1406,10 @@ def get_extensions(self, r=None, basename='', atm_ind=None, atm_ind2=None, n_spl
...
+        if r_un is None:
+            r_un = [0,1,2,3]
```
Comment (hwpang) [inline rmgpy/molecule/group.py:1412]:
```suggestion
            r_un = [0, 1, 2, 3]
```

**Review by hwpang — CHANGES_REQUESTED** [review]
A little more changes

**Comment by mjohnson541** [comment]
Okay, I've added those changes. Think this should be ready.

**Review by hwpang — APPROVED** [review]
LGTM
